src/SqlConnectionConfig.ts

```diff
@@ -67,12 +68,138 @@ export default class SqlConnectionConfig {
+                }
+                if (!auth.options?.password) {
+                    throw new Error(`Invalid connection string. Please ensure 'Password' is provided in the connection string.`);
+                }
+                break;
+            }
+            case 'azure-active-directory-service-principal-secret': {
+                if (!auth.options?.clientId) {
+                    throw new Error(`Invalid connection string. Please ensure client ID is provided in the 'User' or 'User ID' field of the connection string.`);
+                }
+                if (!auth.options?.clientSecret) {
+                    throw new Error(`Invalid connection string. Please ensure client secret is provided in the 'Password' field of the connection string.`);
+                }
+                break;
+            }
         }
+    }
+
+    /**
+     * Sets the authentication option in the mssql config object based on the connection string.
+     * node-mssql currently ignores authentication when parsing the connection string: https://github.com/tediousjs/node-mssql/issues/1400
+     * Assumes _connectionConfig has already been set, sets authentication to _connectionConfig directly
+     */
+    private _setAuthentication(): void {
+
+        // Parsing logic from SqlConnectionStringBuilder._parseConnectionString https://github.com/Azure/sql-action/blob/7e69fdc44aba3f05fd02a6a4190841020d9ca6f7/src/SqlConnectionStringBuilder.ts#L70-L128
+        const result = Array.from(this._connectionString.matchAll(Constants.connectionStringParserRegex));
+
+        // TODO: Change this to Array.from().find() now that we're only looking for authentication
+        const authentication = this._findInConnectionString(result, 'authentication');
+        if (!authentication) {
+            // No authentication set in connection string
+            return;
+        }
+
+        // Possible auth types from connection string: https://docs.microsoft.com/sql/connect/ado-net/sql/azure-active-directory-authentication
+        // Auth definitions from tedious driver: http://tediousjs.github.io/tedious/api-connection.html
+        switch (authentication.replace(/\s/g, '').toLowerCase()) {
+            case 'sqlpassword': {
+                // default: use user and password
+                break;
+            }
+            case 'activedirectorypassword': {
+                this._connectionConfig['authentication'] = {
+                    "type": 'azure-active-directory-password',
+                    "options": {
+                      // User and password should have been parsed already  
+                      "userName": this._connectionConfig.user,
+                      "password": this._connectionConfig.password,
+                      "clientId": core.getInput('client-id') || undefined,
+                      "tenantId": core.getInput('tenant-id') || undefined
```

Comment:
@dzsquared do you know if client-id or tenant-id can be exposed to log? Or should we mask them as secrets? passwords and secrets are masked already

Reply:
I masked them just in case

Reply:
Minimally they're identifying information, but could be aggregated with other information to facilitate attack so let's keep them masked.  thank you!